tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,945 @@
+import collections
+import logging
+import base64
+import time
+import os
+from os.path import join
+import tempfile
+import re
+
+import pytest
+
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_ocs_version,
+    ignore_leftovers,
+    skipif_openshift_dedicated,
+    skipif_external_mode,
+    system_test,
+)
+from ocs_ci.ocs.ocp import OCP, switch_to_project
+from ocs_ci.framework.testlib import E2ETest, config
+from ocs_ci.ocs.exceptions import CommandFailed, ResourceWrongStatusException
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility import templating
+from ocs_ci.ocs.resources.pod import (
+    get_mon_pods,
+    get_osd_pods,
+    get_deployments_having_label,
+    get_mds_pods,
+    get_cephfsplugin_provisioner_pods,
+    get_rbdfsplugin_provisioner_pods,
+    get_rgw_pods,
+    get_noobaa_pods,
+    get_plugin_pods,
+    get_ceph_tools_pod,
+)
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs import ocp, constants, defaults, bucket_utils
+from ocs_ci.helpers.helpers import wait_for_resource_state
+from ocs_ci.utility.retry import retry
+from ocs_ci.utility.utils import exec_cmd
+
+logger = logging.getLogger(__name__)
+
+
+@system_test
+@ignore_leftovers
+@pytest.mark.polarion_id("OCS-3911")
+@pytest.mark.bugzilla("1973256")
+@skipif_ocs_version("<4.6")
+@skipif_openshift_dedicated
+@skipif_external_mode
+class TestMonitorRecovery(E2ETest):
+    """
+    Test to verify monitor recovery
+
+    """
+
```

Comment:
I did not find the teardown code, this is a very disruptive procedure and can make the cluster go into bad shape if the procedure fails in middle. Please also consider handling teardown

Reply:
Agree, but there are lot of failure possibilities at different stages, it will be very hard to figure out the point of failure and address it. Alternative for now is that, we would have to skip immediately if this test fails consistently or remove this from tier runs and run it locally every release.
I cant think of a common teardown for this as of now. I feel this whole mon recovery procedure is like a huge teardown to get the cluster running again.  :/

Reply:
added execute last